plugins/nf-validation/src/main/nextflow/validation/SchemaValidator.groovy

```diff
@@ -528,6 +528,13 @@ class SchemaValidator extends PluginExtensionPoint {
                 .primitiveValidationStrategy(PrimitiveValidationStrategy.LENIENT)
                 .build();
             validator.performValidation(schema, arrayJSON);
+            if (this.hasErrors()) {
+                // Needed for custom errors such as pathExists() errors
+                def colors = logColours(monochrome_logs)
+                def msg = "${colors.red}The following errors have been detected:\n\n" + this.getErrors().join('\n').trim() + "\n${colors.reset}\n"
+                log.error("ERROR: Validation of '$paramName' file failed!")
+                throw new SchemaValidationException(msg, this.getErrors())
```

Comment:
Does this mean that it'll exit after the first error?

I remember talking about this a while ago and saying that it could be annoying to keep launching the pipeline again and again to work through a list of errors..

Reply:
It will only exit if the validation passed, but there are other errors. If the validation failed, it will exit with all validation errors + our custom errors.

Reply:
Perfect 🙌🏻